src/main/java/org/folio/auth/authtokenmodule/storage/RefreshTokenStore.java

```diff
@@ -0,0 +1,202 @@
+package org.folio.auth.authtokenmodule.storage;
+
+import org.folio.auth.authtokenmodule.tokens.RefreshToken;
+
+import java.time.Instant;
+import java.util.UUID;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.sqlclient.SqlConnection;
+import io.vertx.sqlclient.Tuple;
+
+/**
+ * The refresh token store provides persistence for refresh tokens. Refresh tokens need
+ * to be persisted so that: 1) we can ensure they are redeemed only once, and 2) we can
+ * take action (revoking all tokens for a user) when a given token is used more than once.
+ * When a refresh token is attempted to be redeemed more than once we consider it leaked.
+ */
+public class RefreshTokenStore extends TokenStore {
+  private static final Logger log = LogManager.getLogger(RefreshTokenStore.class);
+
+  private static String REFRESH_TOKEN_SUFFIX = "refresh_tokens";
+
+  /**
+   * Constructs the refresh token store.
+   * @param vertx A reference to the current vertx object.
+   * @param tenant The tenant which is in scope for a given token. This can be obtained
+   * from the refresh token when the refresh token is created or arrives.
+   */
+  public RefreshTokenStore(Vertx vertx, String tenant) {
+    super(vertx, tenant);
+
+    // TODO implement this to clean up expired tokens from the store.
+    // TODO Consider using an okapi timer.
+    // vertx.setPeriodic(delay, handler)
```

Comment:
I rather prefer this approach:
After a new token has been inserted asynchronously delete expired tokens. No need to run the delete method when there is no activity.